### PR TITLE
feat(Department): New function for Department Management

### PR DESCRIPTION
**Pull request by tuannn2**
Signed-off-by: Nguyen Nhu Tuan <tuan2.nguyennhu@toshiba.co.jp>

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: #1439 

### How To Test?

# Intructions for using the Department Management

- On the Interface, the User can configure the directory path by entering the path and then pressing the update button (Path Forder will save in file config `department-config.json`)
- User tracks on interface with fields: Interval, Last Running Time, Next Running Time
- User schedule it automatically or click the Activate button manually to do the import
- User have to view History about Import Department of System By View Log
- User perfoms edit Department Management like other functions 
[SW360-20220406.-.Demo1439.pptx](https://github.com/eclipse/sw360/files/10236351/SW360-20220406.-.Demo1439.pptx)

# Create File config in folder "Liferay"
Example: /opt/liferay-ce-portal-7.4.3.18-ga18/department-config.json

## Change value in configuration file

1. Open `department-config.json` to edit configuration file

  - `vim department-config.json`
``` sh
    {
    "configDepartment":{
        "lastRunningTime":"${Last Running Time}",
        "pathFolder":"${Folder_path}",
        "showFileLogFrom":${day}
      }
    }

```
    
```sh
    - ${Last Running Time}: Last Running Time of Schedule
    - ${Folder_path}: path Folder contains files Inport,  end with '/' character (e.g: /home/data/)
    - ${day}: Get all log files in last "x" days in a directory and if the date is not set, default get all log files (e.g:10 days, 20 days, 30 days ,...)
```
File Example:
[department-config.json.txt](https://github.com/eclipse/sw360/files/10236278/department-config.json.txt)
- please remove extension "txt"

# Set up Portlet Department
- Import Private Portlet Department in path: **frontend/configuration/Private_Pages_7_4_3_18_GA18.lar**

# File Test Import
- support file csv, xlsx: 
[importUser.xlsx](https://github.com/eclipse/sw360/files/10236344/importUser.xlsx)

- File import success when all email of user existed in System: 

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by KoukiHama** [comment]
Please check these file which @tienlee made for review
https://github.com/eclipse/sw360/files/8440283/SW360-20220406.-.Demo1439.pptx

File: frontend/sw360-portlet/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -1499,3 +1499,18 @@ an.obligation.with.the.same.name.already.exists=An Obligation with the same name
 ## Refer to http://cdn.datatables.net/plug-ins/9dcbecd42ad/i18n/ and add your datatables language
 
 datatables.lang=https://cdn.datatables.net/plug-ins/9dcbecd42ad/i18n/English.json
+schedule.department.service=Schedule Dept. Generating
+cancel.department.service=Cancel Dept. Generating
+member.emails=Member Emails
+manually=Manually Activate
```
Comment (JaideepPalit) [inline frontend/sw360-portlet/src/main/resources/content/Language.properties:1505]: Add same in `Language-vi.properties and Language-jp.properties`

**Review by JaideepPalit — CHANGES_REQUESTED** [review]
Reviewed the PR.
Kindly look into some minor review comments

**Comment by KoukiHama** [comment]
How to set up this function.
https://github.com/toshiba/sw360/wiki/Guide_Department_Management.md

**Comment by tienlee** [comment]
Thank you for your reviewing, @JaideepPalit 
The defects were fixed following your comments, please help me to confirm.

File: backend/src/src-users/src/main/java/org/eclipse/sw360/users/UserHandler.java [hunk 2]
```diff
@@ -137,7 +143,147 @@ public Set<String> getUserDepartments() throws TException {
...
+    public Set<String> getListFileLog() {
+        try {
+            DepartmentConfigDTO configDTO = readFileDepartmentConfig.readFileJson();
+            if (configDTO != null && !configDTO.getPathFolderLog().isEmpty()) {
+                String path = configDTO.getPathFolderLog();
+                File theDir = new File(path);
+                if (!theDir.exists()) theDir.mkdirs();
+                return FileUtil.listFileNames(path);
+            }
+        } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (tuannn2) [reply backend/src/src-users/src/main/java/org/eclipse/sw360/users/UserHandler.java:197]: thank @JaideepPalit. I changed Exception to IOException!

[the remark this answers is not in the document]
Comment (tuannn2) [reply backend/src/src-users/src/main/java/org/eclipse/sw360/users/UserHandler.java:197]: thank @JaideepPalit. I changed Exception to IOException!

**Comment by tuannn2** [comment]
Thank you for your reviewing, @JaideepPalit  . I have fixed it according to your review. Please check it!

**Review by JaideepPalit — APPROVED** [review]
Code looks good to me.

**Comment by KoukiHama** [comment]
@tuannn2 please fix merge conflict

**Comment by tuannn2** [comment]
Thank you @KoukiHama. I have resolved the conflict. Please change the label!

**Comment by tienlee** [comment]
Dear all,

Toshiba's members were updated this PR follow latest code from main branch.
In addition, we also updated the description for "How to test" section. Please check and test this PR.

Thank you very much.

**Comment by nikkuma7** [comment]
Kindly make a lar file for Department portlet.so we can directly import the lar file instead doing the manual changes for department page under admin portlet.

**Comment by nikkuma7** [comment]
image
I'm following as per your given PPT test guide, but Department tab is not showing under the sw360 Administration tab.

**Comment by tuannn2** [comment]
Thank you @nikkuma7,

In the latest commit, I created the file lar.
Please import Private Portlet Department in path: frontend/configuration/Private_Pages_7_4_3_18_GA18.lar in source code, instead of having to setup manual portlet.

**Comment by nikkuma7** [comment]
Yes, I added a private lar file ( which you mentioned above)
image
image

department page is still not showing.

**Comment by nikkuma7** [comment]
yes, I select private and others options like 

1:Import Permissions
2:Mirror with overwriting 
3:use the current user as author
image

could you please let me know your availability on slack so we can discuss there.

**Comment by KoukiHama** [comment]
> could you please let me know your availability on slack so we can discuss there.

If you are discussing this in a private channel on slack, can you make it public?

**Comment by tienlee** [comment]
We are discussing on https://sw360chat.slack.com/archives/C04JENDJGUC

I am busy, so I have not solved the issue with @nikkuma7 (added a private lar file but the department page is still not showing) now.

Sorry for this inconvenience.

**Comment by nikkuma7** [comment]
image

Now department button is showing in admin tab, but still facing the issue (department page is not loading).
please see the below SS.
image

**Comment by nikkuma7** [comment]
Test Successful.
image

**Comment by nikkuma7** [comment]
image

**Comment by KoukiHama** [comment]
we will fix conflict soon.
@tuannn2 @tienlee Could you check it?

**Comment by tuannn2** [comment]
Thank you @KoukiHama. 
I have resolved the conflict. 
Please change the label !

**Comment by tienlee** [comment]
JaideepPalit reviewed this PR and tested by nikkuma7. This PR is already to merge.

**Comment by nikkuma7** [comment]
@tienlee , Again portlet is not loading.facing issue while loading admin -> department page.

image

**Review by ag4ums — APPROVED** [review]
working as expected

**Pull request merged**